src/mlpack/methods/kde/kde_stat.hpp

```diff
@@ -52,6 +54,18 @@ class KDEStat
                            "value");
   }
 
+  //! Get depth of the node.
+  inline size_t Depth() const { return depth; }
+
+  //! Modify depth of the node.
+  inline size_t& Depth() { return depth; }
+
+  //! Get accumulated Monte Carlo alpha of the node.
+  inline double AccumAlpha() const { return accumAlpha; }
```

Comment:
Do we need to ensure that this is set to 0 before traversal?  Someone may call `KDE::Evaluate()` with a query tree that's already been used.

Reply:
You're right. This should be solved with dd0dca90f555c7f6cc58559581e4bf7d197d0903 and 7ea2abfeefd193f71878ac07d1c743e72a47d3f0 :)